cmd/kops/update_cluster.go

```diff
@@ -138,6 +143,22 @@ func RunUpdateCluster(ctx context.Context, f *util.Factory, clusterName string,
 	isDryrun := false
 	targetName := c.Target
 
+	if c.admin && c.user != "" {
+		return nil, fmt.Errorf("cannot use both --admin and --user")
+	}
+
+	if c.CreateKubecfg && !c.admin && c.user == "" {
+		return nil, fmt.Errorf("--create-kube-config requires that either --admin or --user is set")
+	}
+
+	if c.admin && !c.CreateKubecfg {
+		return nil, fmt.Errorf("--admin requires that --create-kube-config is set")
+	}
+
+	if c.user != "" && !c.CreateKubecfg {
+		return nil, fmt.Errorf("--user requires that --create-kube-config is set")
+	}
```

Comment:
That is a good idea! I'll give that a go.

Reply:
The logging probably isn't necessary.

Reply:
I think they could be helpful and I guess common for CLI tools. But we also do document that it is implied. If you think they should go, I can remove it.

Reply:
I think most users aren't going to keep `--create-kube-config` in mind. They're going to know either `--admin` or `--user`, whichever fits their use case. So I think this would just be log chatter.